gitlab/__init__.py

```diff
@@ -568,7 +568,7 @@ def update(self, obj, **kwargs):
         params.update(kwargs)
         missing = []
         for k in itertools.chain(obj.requiredUrlAttrs,
-                                 obj.requiredCreateAttrs):
+                                 obj.requiredUpdateAttrs):
```

Comment:
You need to deal with GitlabObjects that don't have requiredUpdateAttrs defined. For these classes requiredCreateAttrs needs to be used.

This might be why the unit tests are failing.